docs/reference/sql/data-definition.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: page
+title: Data definition
+tagline: Use DDL to structure data 
+description: How to use DDL to structure data in ksqlDB
+keywords: ksqldb, sql, ddl
+---
+
+This section covers how you create the structures that store your events.
+ksqlDB abstracts events as rows with columns and stores them in streams
+and tables.
+
+## Rows and columns
+
+Streams and tables help you model collections of events that accrete over time.
+Both are represented as a series of rows and columns with a schema, much like a
+relational database table. Rows represent individual events. Columns represent
+the attributes of those events.
+
+Each column has a data type. The data type limits the span of permissible values
+that you can assign. For example, if a column is declared as type `INT`, it can't
+be assigned the value of string `'foo'`.
+
+In contrast to relational database tables, the columns of a row in ksqlDB are
+divided into _key_ and _value_ columns. The key columns control which partition
+a row resides in. The value columns, by convention, store the main data of
+interest. Controlling the key columns is useful for manipulating the underlying
+data locality, and enables you to integrate with the wider {{ site.ak }}
+ecosystem, which uses the same key/value data model. By default, a column is a
+value column. Marking a column as a `(PRIMARY) KEY` makes it a key column.
```

Comment:
May be worth also mentioning that they key column also controls ordering guarantees, i.e. only rows in the same partition have any relative ordering guarantees.